R/chk-filter.R

```diff
@@ -74,34 +81,31 @@ chk_filter <- function(data, var, func, flt = TRUE, args = list()) {
 #' @rdname chk-helper
 #' @export
 chk_filter_vars <- function(data, vars, func, flt = TRUE, args = list()) {
```

Comment:
A few things:
* `chk_filter_vars()` is a bit of an odd duckling, returning a data.frame, an undocumented quirk. 
* `chk_filter()` isn't strictly necessary, given that it's equivalent to both `chk_filter_all()` and `chk_filter_any()` in situations where only one variable is mentioned
* `chk_filter_where()` can be hard deprecated. It's now identical to `chk_filter_all()` and probably isn't actually being used by anyone yet.

Reply:
Perhaps there should just be one exported function, `chk_filter()` which behaves as `chk_filter_vars()`? It's not too much trouble for the user to write:

```r
chk_filter(...) %>% apply(1, all)
``` 

Would simplify the documentation greatly

Reply:
Agree re: `chk_filter()`, it's totally redundant now. What do you reckon of renaming `chk_filter_vars()` to `chk_filter()`? The `_vars` seems redundant with the single variable variant gone.

Not so sure about dropping the `_all()` and `_any()` variants, they're pretty handy helpers and simple enough to leave in. I think removing `chk_filter()` and `chk_filter_where()` should simplify the docs enough.

Reply:
Yeah, definitely agree re `chk_filter()` and `chk_filter_vars()`. 

As to retaining the other `chk_filter_*()` variants, my only misgiving would be that we'd still have the slight inconsistency in return type between `chk_filter()` and `chk_filter_*()`. It's not really a very big deal as long as it's documented. But given that the wrappers are so simple, it seems to make sense to just drop them. I doubt any user of `testdat` would have trouble coming up with `apply(chk_filter(...), 1, all)` or `apply(chk_filter(...), 1, any)` on their own should they need them. (Plus, we could point them in the right direction by including an example like this in the function documentation). And I can't see them being used so often that the additional ~10 characters will be much of an inconvenience. 

Happy with whatever you decide

Reply:
For reference, left the `_all()` and `_any()` variants in.